tests/test_helpers.py

```diff
@@ -0,0 +1,63 @@
+import pytest
+import os
+import json
+import pandas as pd
+import tempfile
+from queue import Queue
+from workers.util import read_config
+from workers.worker_base import Worker
+from augur.config import AugurConfig, default_config
+from augur import ROOT_AUGUR_DIRECTORY
+
+temp_dir = os.path.join(os.getcwd(), "util")
+config_path = os.path.join(temp_dir, "test.config.json")
+def test_dump_queues():
+    sample_queue = Queue()
+    list_sample = ["x@x.com", "y@y.com", "z@z.com"]
+    for list_item in list_sample:
+        sample_queue.put(list_item)
+    queue_to_list = Worker.dump_queue(sample_queue)
+    assert queue_to_list == ["x@x.com", "y@y.com", "z@z.com"]
+
+def test_read_config_no_exception():
+    test_config = default_config
+    base_dir = os.path.dirname(os.path.dirname(__file__))
+    print(base_dir)
+    with open(config_path, "w") as f:
+        json.dump(test_config, f)
+    db_name = read_config('Database', 'user', 'AUGUR_DB_USER', 'augur', config_file_path=config_path)
+    assert db_name == "augur"
+
+def test_read_config_exception():
+    with pytest.raises(AttributeError):
+        db_name = read_config('Server', 'username')
+
+def test_config_get_section_no_exception():
+    test_config = default_config
+    test_config['Database']['user'] = "test_user"
+    config_object = AugurConfig(temp_dir, test_config)
+    assert type(config_object.get_section("Database")) == dict
+
+def test_config_get_section_exception():
+    test_config = default_config
+    test_config['Database']['user'] = "test_user"
+    config_object = AugurConfig(temp_dir, test_config)
+    assert config_object.get_section("absent_section") == None
+
+def test_discover_config_file_env_exception():
+    os.environ['AUGUR_CONFIG_FILE'] = os.path.join(temp_dir, "augur.config.json")
+    test_config = default_config
+    with pytest.raises(FileNotFoundError):
+        config_object = AugurConfig(temp_dir, test_config)
+        assert config_object.discover_config_file()
+
+def test_discover_config_file_env_no_exception():
+    test_config = default_config
+    with open(os.path.join(temp_dir, "augur.config.json"), "w") as f:
+        pass
+    config_object = AugurConfig(temp_dir, test_config)
+    assert config_object.discover_config_file() == os.path.join(temp_dir, "augur.config.json")
+
+
+
+
```

Comment:
This file needs to be deleted. These tests break **EVERYTHING** and we do not use them.

Reply:
Can you clarify how they break everything? Worker tests are generally a good thing, and we don't have enough of them. I assume they break everything because they are organized around the way we operated the augur.config.json and the old general worker. Thanks, @ccarterlandis

Reply:
1. These are not worker tests, they have nothing to do with the workers. These are only dealing with the config discovery and generation logic.
2. They are poorly written, which causes them to interfere with other test cases. For one, none of the test clean up after themselves, leaving their temporary directories and files. Because these configs are not deleted, and are written to a poorly chosen location, all the other tests see the temporary config files and use them instead of the actual valid config.
3. It uses old ways of interacting with the config which are not relevant anymore.

I'm not currently interested in rewriting or reworking these tests, so I'd like for them to be removed. This is maybe the 3rd or 4th time they have tried to make their way into `dev`, as they are a consequence of when you merged `gsoc-dev` into `dev` at the end of the summer.

Reply:
Each time there is a PR which would merge them, we've had to delete them. Trust me, you don't want these tests in `dev`.

Reply:
@ccarterlandis : OK, I will remove the tests, and maintain them in a different branch to be written along the lines you describe at a later time. We did work hard on the tests, even though we got it wrong. So, I will declare victory in the attempt!  EVERYONE WINS A TROPHY!

Reply:
@ccarterlandis Thanks for explaining, btw.  I sincerely wanted to understand.

Reply:
I have to fix this yet.